src/components/GameWorld.hpp

```diff
@@ -0,0 +1,301 @@
+#pragma once
+#include <BsPrerequisites.h>
+#include <RTTI/RTTIUtil.hpp>
+#include <Scene/BsComponent.h>
+
+namespace REGoth
+{
+  class Item;
+  using HItem = bs::GameObjectHandle<Item>;
+
+  class Waynet;
+  using HWaynet = bs::GameObjectHandle<Waynet>;
+
+  class Character;
+  using HCharacter = bs::GameObjectHandle<Character>;
+
+  class GameWorld;
+  using HGameWorld = bs::GameObjectHandle<GameWorld>;
+
+  namespace Scripting
+  {
+    class ScriptVMForGameWorld;
+  }
+
+  /**
+   * Component to be attached to the scene root which manages the game world.
+   *
+   * This is the place where ZEN-files are imported, Characters placed into the
+   * world and where the worlds script engine is at home.
+   *
+   *
+   * Loading and Saving Worlds
+   * =========================
+   *
+   * When a world shall be loaded, the GameWorld component needs to be created
+   * and attached to a Scene Object, which we call *Root SO*. By manually adding
+   * the GameWorld component via `addComponent()`, you can specify the name of a
+   * original ZEN-file. This ZEN-file will be parsed and imported into the
+   * scene, where all created scene objects will be a (direct or indirect) child
+   * of the *Root SO*.
+   *
+   * Once a world is imported it will function as a clean starting point. When
+   * running the game, at that point you would want do run the script init
+   * routine, but you don't have to. What happens after the import is done can
+   * be controlled by overriding the `onImportedZEN()` method.
+   *
+   * At any time after the world has been created, it can be saved to a file,
+   * similar to a save-game in the original game. This save game will store the
+   * complete state of the world, including script state.
+   *
+   * Example to create an empty world:
+   *
+   *    HGameWorld gameWorld = GameWorld::createEmpty();
+   *
+   *
+   * Example to import a ZEN:
+   *
+   *    HGameWorld gameWorld = GameWorld::importZEN("OLDWORLD.ZEN");
+   *
+   *
+   * Example to save a World:
+   *
+   *    gameWorld->save("MySavegame");
+   *
+   *
+   * Example to load a saved World:
+   *
+   *    bs::HPrefab prefab = GameWorld::load("MySavegame");
+   *    prefab->instantiate();
```

Comment:
The documentation says that 

> [...] This is done by calling Prefab::instantiate() which returns a SceneObject. By default this scene object will be parented to the current scene root, but can then be manipulated as any other scene object.

What happens if I save the current scene root and perform a load operation directly after? Will the previously saved root be parented to the current scene root (instead of replacing it), resulting in "two" GameWorld components being active at the same time?

Reply:
I think there are special *Scene*-prefabs which will replace the root. But not that I think of it, I remember seeing a commit which changed the way to load scenes from using a prefab to going through the `SceneManager` or what's it called. I will look it up later.

Reply:
There is `gSceneManager().save(...)` but that also saves stuff like the FPS-Camera, which does not have RTTI. We'll figure that out in another PR, if that's okay for you.